src/coreComponents/fileIO/vtk/VTKPolyDataWriterInterface.cpp

```diff
@@ -912,5 +913,21 @@ void VTKPolyDataWriterInterface::write( real64 const time,
   m_previousCycle = cycle;
 }
 
+bool VTKPolyDataWriterInterface::isFieldPlotEnabled( PlotLevel const wrapperPlotLevel,
+                                                     string const & wrapperName ) const
+{
+  // check if the logLevel is sufficient high for plotting
+  bool plotEnabled = ( wrapperPlotLevel <= m_plotLevel );
+
+  // override the logLevel if the fieldNames list was provided
+  if( !m_fieldNames.empty() )
+  {
+    auto search = m_fieldNames.find( wrapperName );
+    plotEnabled = ( search != m_fieldNames.end() );
+  }
+  return plotEnabled;
```

Comment:
It looks like I've seen this code before 🤣

Reply:
Yes, I agree, this code is duplicated. It is not easy to do something better because they live in different folders (`vtk` and `silo`), but I will find something

Reply:
Done, this piece of code was indeed duplicated, but I removed the duplication in the latest commit